tango_ros_common/tango_ros_native/src/tango_ros_node.cpp

```diff
@@ -1013,52 +1083,71 @@ void TangoRosNode::PublishMesh() {
-          LOG(ERROR) << "Tango3DR_GridIndexArray_destroy failed with error code: " << result;
+          LOG(ERROR) << "Tango3DR_GridIndexArray_destroy failed with error code"
+              ": "<< result;
         }
         if (mesh_marker_array.markers.empty()) {
           LOG(INFO) << "Empty mesh array!";
         }
         mesh_marker_publisher_.publish(mesh_marker_array);
       }
-      // Publish Tango mesh as occupancy grid.
-      if (occupancy_grid_publisher_.getNumSubscribers() > 0) {
-        Tango3DR_Status result  = Tango3DR_updateFullFloorplan(t3dr_context_);
-        if (result == TANGO_3DR_SUCCESS) {
-          Tango3DR_Vector2 origin;
-          Tango3DR_ImageBuffer image_grid;
-          result = Tango3DR_extractFullFloorplanImage(
-              t3dr_context_, TANGO_3DR_LAYER_OBSTACLES, &origin, &image_grid);
+      auto get_and_publish_occupancy_grid = [&] (ros::Publisher& publisher,
+          Tango3DR_FloorplanLayer layer) {
+        // Publish Tango mesh as occupancy grid.
+        if (publisher.getNumSubscribers() > 0) {
+          Tango3DR_Status result  = Tango3DR_updateFullFloorplan(t3dr_context_);
           if (result == TANGO_3DR_SUCCESS) {
-            nav_msgs::OccupancyGrid occupancy_grid;
-            tango_ros_conversions_helper::toOccupancyGrid(
-                image_grid, origin, time_offset_, t3dr_resolution_, &occupancy_grid);
-            occupancy_grid_publisher_.publish(occupancy_grid);
+            Tango3DR_Vector2 origin;
+            Tango3DR_ImageBuffer image_grid;
+            result = Tango3DR_extractFullFloorplanImage(
+                t3dr_context_, layer, &origin, &image_grid);
+            if (result == TANGO_3DR_SUCCESS) {
+              nav_msgs::OccupancyGrid occupancy_grid;
+              tango_ros_conversions_helper::toOccupancyGrid(
+                  image_grid, origin, time_offset_, t3dr_resolution_,
+                  &occupancy_grid);
+              publisher.publish(occupancy_grid);
+            } else {
+              LOG(ERROR) << "Tango3DR_extractFullFloorplanImage failed with"
+                  " error code: " << result;
+            }
+            result = Tango3DR_ImageBuffer_destroy(&image_grid);
+            if (result != TANGO_3DR_SUCCESS) {
+              LOG(ERROR) << "Tango3DR_ImageBuffer_destroy failed with error"
+                  " code: " << result;
+            }
           } else {
-            LOG(ERROR) << "Tango3DR_extractFullFloorplanImage failed with error code: " << result;
-          }
-          result = Tango3DR_ImageBuffer_destroy(&image_grid);
-          if (result != TANGO_3DR_SUCCESS) {
-            LOG(ERROR) << "Tango3DR_ImageBuffer_destroy failed with error code: " << result;
+            LOG(ERROR) << "Tango3DR_updateFullFloorplan failed with error"
+                " code: " << result;
           }
-        } else {
-          LOG(ERROR) << "Tango3DR_updateFullFloorplan failed with error code: " << result;
         }
-      }
+      };
```

Comment:
Converting above function to a private helper function would make the code easier to read, but this is just my opinion.